packages/xchain-bitcoincash/src/utils.ts

```diff
@@ -290,18 +285,23 @@ export const buildTx = async ({
     // Outputs
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     outputs.forEach((output: any) => {
+      let out = undefined
       if (output.script) {
-        transactionBuilder.addOutput(compiledMemo, 0) // Add OP_RETURN {script, value}
+        out = compiledMemo
       } else if (!output.address) {
         //an empty address means this is the  change address
-        const out = bitcash.address.toOutputScript(toLegacyAddress(sender), bchNetwork(network))
-        transactionBuilder.addOutput(out, output.value)
+        out = bitcash.address.toOutputScript(toLegacyAddress(sender), bchNetwork(network))
       } else if (output.address) {
-        const out = bitcash.address.toOutputScript(toLegacyAddress(output.address), bchNetwork(network))
-        transactionBuilder.addOutput(out, output.value)
+        out = bitcash.address.toOutputScript(toLegacyAddress(output.address), bchNetwork(network))
       }
+      transactionBuilder.addOutput(out, output.value)
     })
 
+    // add output for memo
+    if (compiledMemo) {
+      transactionBuilder.addOutput(compiledMemo, 0) // Add OP_RETURN {script, value}
```

Comment:
is this adding the memo twice? I see on line 290 out is also being set as compiled memo?

Reply:
thanks for catching this !